### PR TITLE
Limit the number of decimal place

### PR DESCRIPTION
**Pull request by iketheadore**
- Only one decimal place is allowed
- Replace visor.StrToBalance() with droplet.FromString()

### CODE REVIEW

File: src/api/cli/create_rawtx.go [hunk 1]
```diff
@@ -464,15 +464,17 @@ func getSufficientUnspents(unspents []UnspentOut, amt uint64) ([]UnspentOut, err
 // NewTransaction create skycoin transaction.
 func NewTransaction(utxos []UnspentOut, keys []cipher.SecKey, outs []coin.TransactionOutput) (*coin.Transaction, error) {
 	tx := coin.Transaction{}
-	// keys := make([]cipher.SecKey, len(utxos))
 	for _, u := range utxos {
 		tx.PushInput(cipher.MustSHA256FromHex(u.Hash))
 	}
 
 	for _, o := range outs {
+		if o.Coins%1e5 != 0 {
```
Comment (gz-c) [inline src/api/cli/create_rawtx.go]: `1e5` should be in a const variable at the top of the file or imported from a single place in a skycoin/src/ package

File: src/api/cli/create_rawtx.go [hunk 2]
```diff
@@ -464,15 +464,17 @@ func getSufficientUnspents(unspents []UnspentOut, amt uint64) ([]UnspentOut, err
...
+			return nil, errors.New("invalid amount, too manay decimal places")
```
Comment (gz-c) [inline src/api/cli/create_rawtx.go]: manay -> many

File: src/daemon/visor.go [hunk 3]
```diff
@@ -346,6 +346,13 @@ func (vs *Visor) verifyTransaction(txn coin.Transaction) error {
 		return fmt.Errorf("Transaction Verification Failed, %v", err)
 	}
 
+	// check the coin decimal place
+	for _, out := range txn.Out {
+		if out.Coins%1e5 != 0 {
```
Comment (gz-c) [inline src/daemon/visor.go]: same const to move, maybe droplet package

File: src/visor/readable.go [hunk 4]
```diff
@@ -242,14 +187,37 @@ func (os ReadableOutputSet) SpendableOutputs() []ReadableOutput {
...
+func NewReadableOutputs(uxs []coin.UxOut) ([]ReadableOutput, error) {
+	rxReadables := make([]ReadableOutput, len(uxs))
+	for i, ux := range uxs {
+		out, err := NewReadableOutput(ux)
+		if err != nil {
+			return []ReadableOutput{}, err
+		}
+
+		rxReadables[i] = out
+		if ux.Hash().Hex() == "f811852ece9932e8c5c0e15121a8c016eb29c2bd98b3c834705b986e9ac1c646" {
```
Comment (gz-c) [inline src/visor/readable.go]: remove?

Comment (iketheadore) [reply src/visor/readable.go]: yes, lol, this was used for debugging.

**Pull request merged**